### PR TITLE
Fixes #42 Refactor to expose functionality to parse _bulk_get respons…

### LINKED ISSUE #42: Refactor to expose functionality to parse _bulk_get response into doc slice
[sgload](http://github.com/couchbaselabs/sgload) is using sg-replicate as a dependency, and needs to use this functionality to handle _bulk_get responses.

Ideally the common code in sgload and sg-replicate would be factored into a common client library.

### PR DESCRIPTION
**Pull request by tleyden**
…e into doc slice

https://github.com/couchbaselabs/sg-replicate/issues/42

### CODE REVIEW

**Comment by tleyden** [comment]
Note that I did have to convert some logging statements so that they will no longer emit the replication ID.  The good news is that this code can avoid being duplicated, the bad news is that now the logging is less useful.  I think we should figure out a way to get both somehow.

File: attachment.go [hunk 1]
```diff
@@ -19,15 +19,15 @@ func NewAttachment(r *Replication, part *multipart.Part) (*Attachment, error) {
 	// copy headers into Headers
 	contentType := part.Header["Content-Type"][0]
 	contentDisposition := part.Header["Content-Disposition"][0]
-	r.LogTo("Replicate", "attachment contentType: %v", contentType)
-	r.LogTo("Replicate", "attachment contentDisposition: %v", contentDisposition)
+	LogTo("Replicate", "attachment contentType: %v", contentType)
+	LogTo("Replicate", "attachment contentDisposition: %v", contentDisposition)
```
Comment (adamcfraser) [inline attachment.go]: I think that including the replication context in the output is valuable - I wouldn't want to just drop it completely.  I'm thinking the approach would be something like https://github.com/couchbaselabs/sg-replicate/pull/28#issuecomment-248752960 - what do you think of that approach?

File: logging.go [hunk 2]
```diff
@@ -25,9 +26,14 @@ func prefixWithReplicationId(lr LoggingReplication, raw string) string {
 	replicationId := lr.Parameters.ReplicationId
 
 	if replicationId != "" {
-		return strings.Join([]string{"[",replicationId ,"] ", raw}, "")
+		return strings.Join([]string{"[", replicationId, "] ", raw}, "")
```
Comment (adamcfraser) [inline logging.go]: We need to come up with a stricter punishment for merges that don't run baseline go fmt - really want to lose this noise.

Comment (tleyden) [inline logging.go]: Yep, agreed.  Also in a separate PR, we should run every file in this repo through `go fmt`

**Comment by tleyden** [comment]
> I agree - the functional requirement here is all about contextual logging, so it probably makes more sense to pass in some kind of Logger interface (which Replication could be an implementation of).

Yeah, I think that's probably the most practical approach right now.  

Closing PR, will re-open once I get those changes in

**Pull request merged**